pennylane/ops/qubit/hamiltonian.py

```diff
@@ -203,6 +203,11 @@ def __init__(
             )
 
         for obs in observables:
+            if isinstance(obs, qml.ops.Prod):
+                # we don't really want this to ever happen, but it makes testing easier during the deprecation phase
+                # because we don't have to generate separate lists of parametrization for old and new opmath testing
+                obs = qml.simplify(obs)
+                obs = qml.operation.Tensor(*obs)
```

Comment:
this, I think, is the only thing that happens in this PR that has any potential for controversy

Reply:
This will be removed once legacy opmath is fully deprecated. Is that so?

Reply:
I guess the whole thing will go, yes. In the mean time, I think this is harmless and will rarely come up. If people give a prod to a legacy Hamiltonian, it's not that unexpected that it would become a Tensor. I think this is a strictly positive change.

Reply:
That being said I am surprised to see how slow `LinearCombination` is 😬 but thats a different topic

```
ops = [X(i) @ X(i+1) for i in range(10)]
ops += [Y(i) @ Y(i+1) for i in range(10)]
ops += [Z(i) @ Z(i+1) for i in range(10)]
coeffs = np.ones(30, dtype=float)

%timeit qml.ops.LinearCombination(coeffs, ops)
```

```
404 µs ± 14.1 µs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)
```

Reply:
The more relevant comparison is `master` and this branch with new opmath disabled. I see a `~4.2%` regression from just the instance check for every operator.

Reply:
@Qottmann it will be a bit tedious and I am very tired of this PR, but you have convinced me 😞 

I'll fix the remaining failing tests so nothing gets mixed up, and then I'll remove this and go back and fix the test-case generation in whatever tests fail as a result so this isn't needed.

Reply:
This is removed now, and the relevant tests are adjusted to accomodate :)